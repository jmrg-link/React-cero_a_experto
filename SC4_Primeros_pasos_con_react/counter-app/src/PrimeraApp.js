import React, { Fragment } from "react";
import PropTypes from 'prop-types'

// FC
const PrimeraApp = ( { saludo , subtitulo} ) => {

  
  return (
    <>
      <h1>{ saludo }</h1>
      <p>{ subtitulo }</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
  
}
PrimeraApp.defaultProps = {
  subtitulo:'Soy un parrado'
}

export default PrimeraApp;
