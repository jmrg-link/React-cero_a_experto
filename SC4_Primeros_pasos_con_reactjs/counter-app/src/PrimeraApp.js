import React, { Fragment } from 'react';

// Funcional Component
const PrimeraApp =  ( { saludo = 'hola mundo' } ) => {


    


    return (
        <Fragment >
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
            <p>Mi primer controlador</p>
        </Fragment>
    )
}

export default PrimeraApp