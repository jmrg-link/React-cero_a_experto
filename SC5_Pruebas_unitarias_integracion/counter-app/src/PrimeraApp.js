import React, { Fragment } from 'react';
import { PropTypes } from "prop-types";

// Funcional Component
const PrimeraApp =  ( { 
    saludo,
    subtitle
} ) => {


    return (
        <Fragment >
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
            <p>{subtitle}</p>
        </Fragment>
    )
}

PrimeraApp.prototype = {
    saludo: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    saludo: 'Hola a todos',
    subtitle:'hola subtitulo'
}

export default PrimeraApp