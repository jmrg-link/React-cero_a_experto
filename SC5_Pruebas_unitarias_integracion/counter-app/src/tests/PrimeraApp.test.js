//import { render  } from '@testing-library/react'
import React from 'react';
import PrimeraApp from '../PrimeraApp'
import { shallow } from 'enzyme'
import { createSerializer } from 'enzyme-to-json'
import '@testing-library/jest-dom'

describe('Pruebas en PrimeraApp', () => { 
    test('debe de mostrar <PrimeraApp /> saludo correctamente', () => { 
        const saludo = 'Hola a todos'
        const wrapper = shallow(<PrimeraApp/>)
        expect(wrapper).toMatchSnapshot()
     })

    test('debe de mostrar <PrimeraApp /> subtitle correctamente', () => { 
        const subtitle = 'hola subtitulo'
        const saludo = 'Hola a todos'
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitle={subtitle}
                />)
        const textP = wrapper.find('p').text()
        console.log(textP);
        expect( textP ).toBe(subtitle)
        
     })
    
    //test('debe de mostrar el mensaje de title', () => { 
    //        const saludo = 'Hola a todos'
    //        const { getByText } = render( <PrimeraApp /> )
    //        // eslint-disable-next-line testing-library/prefer-screen-queries
    //        expect( getByText(saludo)).toBeInTheDocument()
    // })

 })