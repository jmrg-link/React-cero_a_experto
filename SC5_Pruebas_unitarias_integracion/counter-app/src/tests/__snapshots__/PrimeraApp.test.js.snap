// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe de mostrar <PrimeraApp /> saludo correctamente 1`] = `
<Fragment>
  <h1>
    Hola a todos
  </h1>
  <p>
    hola subtitulo
  </p>
</Fragment>
`;
